ui/app/templates/vault/cluster/access/namespaces/index.hbs

```diff
@@ -35,32 +35,51 @@
           {{list.item.id}}
         </Item.content>
         <Item.menu>
-          {{#let (concat this.currentNamespace (if this.currentNamespace "/") list.item.id) as |targetNamespace|}}
-            {{#if (includes targetNamespace this.accessibleNamespaces)}}
-              <li class="action">
-                <NamespaceLink @targetNamespace={{targetNamespace}} @unparsed={{true}} @class="is-block">
-                  Switch to Namespace
-                </NamespaceLink>
-              </li>
-            {{/if}}
-          {{/let}}
-          <ConfirmAction
-            @isInDropdown={{true}}
-            @buttonText="Delete"
+          <Hds::Dropdown @isInline={{true}} as |dd|>
+            <dd.ToggleIcon
+              @icon="more-horizontal"
+              @text="Namespaces nav menu"
+              @hasChevron={{false}}
+              data-test-popup-menu-trigger
+            />
+            {{#let (concat this.currentNamespace (if this.currentNamespace "/") list.item.id) as |targetNamespace|}}
+              {{#if (includes targetNamespace this.accessibleNamespaces)}}
+                <dd.Generic>
+                  <NamespaceLink
+                    @targetNamespace={{targetNamespace}}
+                    @unparsed={{true}}
+                    @class="hds-link-standalone hds-link-standalone--size-medium hds-link-standalone--color-primary"
+                  >
+                    <span class="hds-link-standalone__text">
+                      Switch to Namespace
+                      <Icon @name="external-link" />
+                    </span>
+                  </NamespaceLink>
+                </dd.Generic>
+              {{/if}}
+            {{/let}}
+            <dd.Interactive @text="Delete" @color="critical" {{on "click" (fn (mut this.namespaceToDelete) list.item)}} />
+          </Hds::Dropdown>
+        </Item.menu>
+        {{! Modal inside <ListItem> because we need access to Item.callMethod }}
+        {{#if this.namespaceToDelete}}
+          <ConfirmModal
+            @color="critical"
             @confirmTitle="Delete this namespace?"
             @confirmMessage="Any engines or mounts in this namespace will also be removed."
-            @onConfirmAction={{action
+            @onClose={{fn (mut this.namespaceToDelete) null}}
+            @onConfirm={{action
               (perform
                 Item.callMethod
```

Comment:
I know 😭

Reply:
Moved the other occurrences of this outside ListView and defined the task in the relative controller file because tests were mysteriously failing. Left this one because `refreshNamespaceList` wasn't working properly and this PR already has too much going on and just needs to end 😭